.github/workflows/styler-actions.yml

```
@@ -0,0 +1,93 @@
+    name: style
+    runs-on: macOS-latest
+    steps:
+      - id: file_changes
+        uses: trilom/file-changes-action@v1.2.3
+      - name: testing
+        run: echo '${{ steps.file_changes.outputs.files_modified}}'   
+      - uses: actions/checkout@v2
+      - uses: r-lib/actions/pr-fetch@master
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - uses: r-lib/actions/setup-r@master
+      - name: Install dependencies
+        run: |
+          Rscript -e 'install.packages("styler")'
+          Rscript -e 'install.packages("devtools")'
+          Rscript -e 'devtools::install_version("roxygen2", version = "7.0.2", repos = "http://cran.us.r-project.org")'
+      - name: string operations
+        run: |
+          echo '${{ steps.file_changes.outputs.files_modified}}' > names.txt
+          cat names.txt | tr -d '[]' > new.txt
+          text=$(cat new.txt)
+          IFS=',' read -ra ids <<< "$text"
+          for i in "${ids[@]}"; do if [[ "$i" == *.R\" || "$i" == *.Rmd\" ]]; then echo "$i" >> new2.txt; fi; done
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v1
+        with:
+          name: artifacts
+          path: new2.txt
+      - name: Style
+        run: for i in $(cat new2.txt); do Rscript -e "styler::style_file("$i")"; done
+      - name: commit
+        run: |
+          git add \*.R
+          git add \*.Rmd
+          git commit -m 'Style'
+      - uses: r-lib/actions/pr-push@master
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          
+          
+  check:
+    needs: [style]
+    runs-on: ubuntu-latest
+    container: pecan/depends:develop
+    steps:
+      - name: check git version
+        id: gitversion
+        run: |
+          v=$(git --version | grep -oE '[0-9\.]+')
+          v='cat(numeric_version("'${v}'") < "2.18")'
+          echo "##[set-output name=isold;]$(Rscript -e "${v}")"
+      - name: upgrade git if needed
+        # Hack: actions/checkout wants git >= 2.18, rocker 3.5 images have 2.11
+        # Assuming debian stretch because newer images have git >= 2.20 already
+        if: steps.gitversion.outputs.isold == 'TRUE'
+        run: |
+          echo 'deb http://deb.debian.org/debian stretch-backports main' >> /etc/apt/sources.list
+          apt-get update
+          apt-get -t stretch-backports upgrade -y git
+      - uses: actions/checkout@v2
+      - uses: r-lib/actions/pr-fetch@master
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - uses: r-lib/actions/setup-r@master
+      - name : download artifacts
+        uses: actions/download-artifact@v1
+        with:
+          name: artifacts
+      - name : make
+        shell: bash
+        run: |
+          cut -d / -f 1-2 artifacts/new2.txt | tr -d '"' > new.txt
+          cat new.txt
+          sort new.txt | uniq > uniq.txt
+          cat uniq.txt      
+          for i in $(cat uniq.txt); do make .doc/${i}; done
```

as above, descriptive filenames could remove a lot of the mystery here, e.g.

```suggestion
          cut -d / -f 1-2 artifacts/new2.txt | tr -d '"' > changed_dirs.txt
          cat changed_dirs.txt
          sort changed_dirs.txt | uniq > needs_documenting.txt
          cat needs_documenting.txt      
          for i in $(cat needs_documenting.txt); do make .doc/${i}; done
```